File: library/src/main/java/leavesc/hello/basis/view/BaseActivity.java
package leavesc.hello.basis.view;

import android.content.Context;
import androidx.appcompat.app.AppCompatActivity;

import leavesc.hello.basis.widget.LoadingDialog;

/**
 * 作者：leavesC
 * 时间：2019/3/23 12:07
 * 描述：
 * GitHub：https://github.com/leavesC
 * Blog：https://www.jianshu.com/u/9df45b87cfdf
 */
public class BaseActivity extends AppCompatActivity implements IBaseViewAction {

    protected LoadingDialog loadingDialog;

    @Override
    public void showLoadingDialog(String message, boolean cancelable) {
        if (loadingDialog == null) {
            loadingDialog = new LoadingDialog(getContext());
        }
        loadingDialog.show(message, cancelable, false);
    }

    @Override
    public void dismissLoadingDialog() {
        if (loadingDialog != null && loadingDialog.isShowing()) {
            loadingDialog.dismiss();
        }
    }

    @Override
    public Context getContext() {
        return this;
    }

}

File: basis/src/main/java/leavesc/hello/basis/view/BaseActivity.kt
package leavesc.hello.basis.view

import android.content.Context
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.DialogFragment
import leavesc.hello.basis.dialog.LoadingDialog

/**
 * 作者：leavesC
 * 时间：2019/6/21 10:47
 * 描述：
 */
open class BaseActivity : AppCompatActivity(), IBaseViewAction {

    override val context: Context
        get() = this

    private var loadingDialog: LoadingDialog? = null

    override fun showLoadingDialog(cancelable: Boolean) {
        if (loadingDialog == null) {
            loadingDialog = LoadingDialog(context)
        }
        loadingDialog?.start(cancelable, cancelable)
    }

    override fun dismissLoadingDialog() {
        loadingDialog?.apply {
            if (isShowing) {
                dismiss()
            }
        }
    }

    protected fun showDialog(dialogFragment: DialogFragment) {
        dialogFragment.show(supportFragmentManager, dialogFragment.javaClass.canonicalName)
    }

}